dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/DefaultAppStartTimeProvider.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.os.Process
+import android.os.SystemClock
+import com.datadog.android.core.internal.system.BuildSdkVersionProvider
+import com.datadog.android.core.internal.system.DefaultBuildSdkVersionProvider
+import java.util.concurrent.TimeUnit
+
+internal class DefaultAppStartTimeProvider(
+    buildSdkVersionProvider: BuildSdkVersionProvider = DefaultBuildSdkVersionProvider()
```

Comment:
if you're only doing this for tests there's a way in which you can avoid that and use directly the `Build.Version` static value inside. We have an extension class that we use to test these things: `ApiLevelExtension` class. Check for its usages and you will see it is pretty straightforward to use it.

Reply:
`ApiLevelExtension` is not preferred, because it is using reflection, and with reflection after some time calls may be ignored for `static final` fields (which `Build.Version.SDK_INT` is), because such fields may be optimized (inlined I guess) by JIT.